piptools/scripts/compile.py

```diff
@@ -61,6 +67,12 @@ def _get_default_option(option_name: str) -> Any:
     is_flag=True,
     help="Clear any caches upfront, rebuild from scratch",
 )
+@click.option(
+    "--extra",
+    "extras",
+    multiple=True,
+    help="names of extras_require to install",
```

Comment:
1) `s/n/N`
2) If this is supposed to install only extras but not the default runtime deps, can this be renamed into `--extra-only`?
3) If it's supposed to merge those extras with the main runtime deps list, can we add an additional flag `--extras-only`/`--exclude-default-deps` or something?

Reply:
It's actually a long-standing feature request in pip and I think there's been no opposition, just nobody had time to come up with an implementation. The main motivation was that people often supply test/dev/docs extras for different processes they have and some of them don't really require the dist itself to be installed. Of course, there's another camp saying that this sort of deps should be put elsewhere because they aren't necessary for the runtime and there should be env pins anyway.
But if pip-tools were to implement this flag, it'd facilitate a hybrid of these two approaches with the direct open-ended deps in the dist metadata configs and the pins existing in separate constraints files.
But yes, this could probably be brought up in a separate discussion.

Reply:
I'm from that camp too. I have a separate environment for every tool. However, most of the tools still need dependencies: `pytest`  for sure, `mypy` for type-checking of third-party libs usage, `pylint` for inference-based tasks. In my projects, only `flake8` works nicely without knowing the project dependencies. And I keep a separate `requirements.txt` for it. Yep, makes the root of the project a bit messier but works. This is why I made [dephell](https://github.com/dephell/dephell/) a long time ago, to handle all this zoo :)